--- rDock_2021.1_src.GADI/src/exe/rbcavity.cxx	2014-05-05 20:18:03.441045562 +1000
+++ rDock_2021.1_src.FORBIDDENOK/src/exe/rbcavity.cxx	2020-06-16 18:18:05.912384000 +1000
@@ -143,7 +143,8 @@
 	}
 	else {
 		cout << "Missing receptor parameter file name"<<endl;
-		poptPrintUsage(optCon, stderr, 0);
+		//SRC poptPrintUsage(optCon, stderr, 0);
+		PrintUsage();
 		exit(1);
 	}
 	// writing command line arguments
--- rDock_2021.1_src.GADI/src/exe/rbdock.cxx	2014-06-13 21:05:52.299269185 +1000
+++ rDock_2021.1_src.FORBIDDENOK/src/exe/rbdock.cxx	2020-06-16 18:17:26.712491000 +1000
@@ -181,7 +181,8 @@
 	// input ligand file, receptor and parameter is compulsory
 	cout << endl << "Command line args:" << endl;
 	if(!inputFile || !receptorFile || !protocolFile) { // if any of them is missing
-		poptPrintUsage(optCon, stderr, 0);
+		//SRC poptPrintUsage(optCon, stderr, 0);
+		PrintUsage();
 		exit(1);
 	} else {
 		strLigandMdlFile	= inputFile;
--- rDock_2021.1_src.GADI/build/makeprogs	2020-07-15 19:45:45.000000000 +1000
+++ rDock_2021.1_src.FORBIDDENOK/build/makeprogs	2021-07-19 17:28:14.438992382 +1000
@@ -10,7 +10,7 @@
 CC		= $(TMAKE_CC)
 CFLAGS		= $(TMAKE_CFLAGS_CONFIG)
 CXX		= $(TMAKE_CXX)
-CXXFLAGS	= $(TMAKE_CXXFLAGS_CONFIG) $(DEFINES) -I../include -I../include/GP -L$(LIBDIR)
+CXXFLAGS	= $(TMAKE_CXXFLAGS_CONFIG) $(DEFINES) -I../include -I../include/GP -L$(LIBDIR) -L$(HOME)/.local/lib -I$(HOME)/.local/include
 #
 # Extra include and lib directories for Daylight exes
 #
